File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int N = nums.length;
        int k = 1;
        int prev = -101;
        for(int i = 1; i < N; i++){
            if (nums[i] != nums[i-1] && nums[i] > nums[k-1]){
                prev = nums[k];
                int temp = nums[i];
                nums[i] = nums[k];
                
                nums[k] = temp;
                
                k++;
            }
            
        }
        return k;
    }
}

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.c
int removeDuplicates(int* nums, int numsSize){
    
    int boundary = 0;
    
    for (int i = 0; i < numsSize; ++i){
        int curNum = nums[i];
        if(boundary == 0 || curNum != nums[boundary - 1]){
            nums[boundary] = curNum;
            boundary++;
        }
    }
    
    return boundary;
    
}